kernels/window.cu

```diff
@@ -0,0 +1,50 @@
+#include <cusp/window.cuh>
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include <complex>
+
+namespace cusp {
+
+template <typename T>
+__global__ void kernel_window(const T *in, T *out, float * window, int window_length, int N) {
+  int i = blockIdx.x * blockDim.x + threadIdx.x;
+  if (i < N) {
+    out[i] = in[i] * window[i%window_length];
+    printf("input element is %f\n", in[i]);
+    printf("window element is %f\n", window[i%window_length]);
+    printf("output element is %f\n", out[i]);
+    printf("\n");
+  }
+}
+
+template <typename T>
+cudaError_t window<T>::launch(const T *in, T *out, float * window, int window_length, int N, int grid_size,
+                                 int block_size, cudaStream_t stream) {
+  if (stream) {
```

Comment:
The if statement can be taken out here because if stream is null, stream=0 corresponds to the default stream and the two statements are equivalent.  So keep the statement in the if() and kill the else()